src/nvim/tui/input.c

```diff
@@ -282,27 +283,44 @@ static bool handle_focus_event(TermInput *input)
 
 static bool handle_bracketed_paste(TermInput *input)
 {
-  if (rbuffer_size(input->read_stream.buffer) > 5 &&
-      (!rbuffer_cmp(input->read_stream.buffer, "\x1b[200~", 6) ||
-       !rbuffer_cmp(input->read_stream.buffer, "\x1b[201~", 6))) {
-    bool enable = *rbuffer_get(input->read_stream.buffer, 4) == '0';
-    // Advance past the sequence
-    rbuffer_consumed(input->read_stream.buffer, 6);
-    if (input->paste_enabled == enable) {
+  RBuffer *rbuf = input->read_stream.buffer;
+
+  if (rbuffer_size(rbuf) > 5
+      && (!rbuffer_cmp(rbuf, "\x1b[200~", 6)
+          || !rbuffer_cmp(rbuf, "\x1b[201~", 6))) {
+    bool enable = *rbuffer_get(rbuf, 4) == '0';
+    rbuffer_consumed(rbuf, 6);  // Advance past the sequence
+
+    if (enable && input->paste_started) {
+      // Bogus "paste start"; forward it.
+      enqueue_input(input, "<C-v><Esc>200~", sizeof("<C-v><Esc>200~") - 1);
```

Comment:
Why not simply `return false` and let the cycle handle the rest? You in any case need some kind of escaping for special characters there.